cassandra-dac/src/test/java/org/sunbird/cassandra/CassandraMockTest.java

```diff
@@ -0,0 +1,643 @@
+    when(selectQuery.where()).thenReturn(where);
+    when(metadata.getKeyspace("sunbird")).thenReturn(keyspaceMetadata);
+
+    when(cluster.connect(Mockito.anyString())).thenReturn(session);
+
+    boundStatement = PowerMockito.mock(BoundStatement.class);
+    PowerMockito.whenNew(BoundStatement.class)
+        .withArguments(Mockito.any(PreparedStatement.class))
+        .thenReturn(boundStatement);
+    when(session.execute(boundStatement)).thenReturn(resultSet);
+    when(session.prepare(Mockito.anyString())).thenReturn(statement);
+
+    when(selectSelection.all()).thenReturn(selectBuilder);
+    when(selectBuilder.from(Mockito.anyString(), Mockito.anyString())).thenReturn(selectQuery);
+    when(session.execute(selectQuery)).thenReturn(resultSet);
+    when(session.execute(boundStatement.bind("123"))).thenReturn(resultSet);
+
+    ColumnDefinitions cd = PowerMockito.mock(ColumnDefinitions.class);
+    String str = "qwertypower(king";
+    when(resultSet.getColumnDefinitions()).thenReturn(cd);
+    when(cd.toString()).thenReturn(str);
+    when(str.substring(8, resultSet.getColumnDefinitions().toString().length() - 1))
+        .thenReturn(str);
+  }
+
+  @Test
+  public void testCassandraConnectionWithoutUserNameAndPassword() throws Exception {
+
+    boolean bool = connectionManager.createConnection(host, port, null, null, cassandraKeySpace);
+    assertEquals(true, bool);
+  }
+
+  @Test
+  public void testCassandraConnectionWithUserNameAndPassword() throws Exception {
+
+    Boolean bool =
+        connectionManager.createConnection(host, port, "cassandra", "password", cassandraKeySpace);
+    assertEquals(true, bool);
+  }
+
+  @Test
+  public void testFailedCassandraConnection() {
+
+    try {
+      connectionManager.createConnection("127.0.0.1", "9042", "cassandra", "pass", "eySpace");
+    } catch (Exception ex) {
+    }
+    assertTrue(500 == ResponseCode.SERVER_ERROR.getResponseCode());
+  }
+
+  @Test
+  public void testCassandraInsertOperation() throws Exception {
+
+    Response response = operation.insertRecord(cassandraKeySpace, "address1", address);
+    assertEquals("SUCCESS", response.get("response"));
+  }
+
+  @Test
+  public void testCassandraInsertOperationFailed() throws Exception {
+
+    PowerMockito.whenNew(BoundStatement.class)
+        .withArguments(Mockito.any(PreparedStatement.class))
+        .thenThrow(
+            new ProjectCommonException(
+                ResponseCode.dbInsertionError.getErrorCode(),
+                ResponseCode.dbInsertionError.getErrorMessage(),
+                ResponseCode.SERVER_ERROR.getResponseCode()));
+
+    Throwable exception = null;
+    try {
+      operation.insertRecord(cassandraKeySpace, "address1", address);
+    } catch (Exception ex) {
+      exception = ex;
+    }
+    assertTrue("DB insert operation failed.".equals(exception.getMessage()));
+  }
+
+  @Test
+  public void testCassandraInsertFailedOperationWithInvalidProperty() throws Exception {
```

Comment:
try to throw error from prominent statement of insert to cassandra. this will give very less line coverage